tracer/src/Datadog.Trace/Debugger/Instrumentation/MethodDebuggerState.cs

```diff
@@ -41,7 +42,8 @@ namespace Datadog.Trace.Debugger.Instrumentation
         /// <param name="startTime">The intended start time of the scope, intended for scopes created in the OnMethodEnd handler</param>
         /// <param name="methodMetadataIndex">The unique index of the method's <see cref="Instrumentation.MethodMetadataInfo"/></param>
         /// <param name="invocationTarget">The current invocation target ('this' object)</param>
-        internal MethodDebuggerState(string probeId, Scope scope, DateTimeOffset? startTime, int methodMetadataIndex, object invocationTarget)
+        /// <param name="condition">The probe condition or null if does not exist</param>
+        internal MethodDebuggerState(string probeId, Scope scope, DateTimeOffset? startTime, int methodMetadataIndex, object invocationTarget, ProbeCondition condition)
```

Comment:
Adding the ProbeCondition here is probably going in the wrong direction, since even though it is not a well supported/tested scenario currently, it should be valid for a method to have several Method Probes with different conditions. And likewise, there could be a line probe with a condition, which again makes things quite confusing when if we store the condition here.